app/models/pawn.rb

```diff
@@ -1,2 +1,42 @@
 class Pawn < Piece
+  def move_legal?(to_row, to_col)
+    return true if single_advance?(to_row, to_col)
+    return true if double_advance?(to_row, to_col)
+    return true if capture_move?(to_row, to_col)
+    return true if en_passant?(to_row, to_col) # need to send signal to capture logic for en passant
+    false
+  end
+
+  def promotion
+  end
+
+  private
+
+  def single_advance?(to_row, to_col)
+    valid_to_row = is_black ? row - 1 : row + 1
+    to_row == valid_to_row && to_col == col
+  end
+
+  def double_advance?(to_row, to_col)
+    valid_to_row = is_black ? 4 : 3
+    to_row == valid_to_row && to_col == col && moves.empty?
+  end
+
+  def capture_move?(to_row, to_col)
+    valid_to_row = is_black ? row - 1 : row + 1
+    to_row == valid_to_row && (to_col - col).abs == 1 && enemy_occupied?(to_row, to_col)
+  end
+
+  def en_passant?(to_row, to_col)
+    valid_to_row = is_black ? row - 1 : row + 1
+    valid_row = is_black ? 4 : 3
+    return false unless moves.length == 1 && row == valid_row
```

Comment:
I think this is incorrect, as what you want to know is did the *other* piece jump on its first move, not did this piece only move once.  Actually I think it might be impossible for moves.length to be 1 and for the piece to be in the valid row.